gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfigurationPropertyTest.java

```diff
@@ -0,0 +1,118 @@
+
+@RunWith(JUnit4.class)
+public class GoogleHadoopFileSystemConfigurationPropertyTest {
+
+    @Test
+    public void testPropertyCreation() {
+        GoogleHadoopFileSystemConfigurationProperty<Integer>
+                NEW_KEY_WITH_DEPRECATED_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", 0, "deprecated.key");
+        assertThat(NEW_KEY_WITH_DEPRECATED_KEY.getKey()).isEqualTo("actual.key");
+
+        GoogleHadoopFileSystemConfigurationProperty<Integer>
+                NEW_KEY_WITHOUT_DEPRECATED_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", 0,null);
+        assertThat(NEW_KEY_WITHOUT_DEPRECATED_KEY.getDefault()).isEqualTo(0);
+    }
+
+    @Test
+    public void testCollectionTypeProperty() {
+        Configuration config = loadConfig("projectId", "serviceAccount", "privateKeyFile");
+        GoogleHadoopFileSystemConfigurationProperty<String>
+                STRING_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", "default-string");
+        GoogleHadoopFileSystemConfigurationProperty<Integer>
+                INTEGER_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                "actual.key", 1);
+        GoogleHadoopFileSystemConfigurationProperty<Collection<String>>
+                COLLECTION_KEY = new GoogleHadoopFileSystemConfigurationProperty<>(
+                        "collection.key", ImmutableList.of("key1", "key2"));
+        assertThrows(IllegalStateException.class, () -> STRING_KEY.getStringCollection(config));
+        assertThrows(IllegalStateException.class, () -> INTEGER_KEY.getStringCollection(config));
+        Collection<String> col = COLLECTION_KEY.getStringCollection(config);
+        assertThat(col).isEqualTo(ImmutableList.of("key1", "key2"));
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testProxyProperties() {
+        GoogleHadoopFileSystemConfigurationProperty<String> GCS_PROXY_USERNAME =
+                new GoogleHadoopFileSystemConfigurationProperty<>(
+                        EntriesCredentialConfiguration.PROXY_USERNAME_KEY,
+                        "proxy-user");
+        GoogleHadoopFileSystemConfigurationProperty<String> GCS_PROXY_PASSWORD =
+                new GoogleHadoopFileSystemConfigurationProperty<>(
+                        EntriesCredentialConfiguration.PROXY_PASSWORD_KEY,
+                        "proxy-pass");
+
+        Configuration config = loadConfig("projectId", "serviceAccount", "privateKeyFile");
+        config.set(GCS_PROXY_USERNAME.getKey(), GCS_PROXY_USERNAME.getDefault());
+        config.set(GCS_PROXY_PASSWORD.getKey(), GCS_PROXY_PASSWORD.getDefault());
+        GoogleCloudStorageFileSystemOptions.Builder optionsBuilder =
+                GoogleHadoopFileSystemConfiguration.getGcsFsOptionsBuilder(config);
+
+        /** Proxy properties should fail when no proxy address is specified */
```

Comment:
Use regular `//` comment, not java doc comment inside the methods (here and below).